scripts/rddepman.ts

```diff
@@ -0,0 +1,145 @@
+  new tools.Trivy(),
+  new tools.GuestAgent(),
+  new tools.Steve(),
+  new tools.RancherDashboard(),
+  new tools.ECRCredHelper(),
+  new LimaAndQemu(),
+  new AlpineLimaISO(),
+  new WSLDistro(),
+  new HostResolverHost(),
+  new HostResolverPeer(),
+  new MobyOpenAPISpec(),
+];
+
+function git(...args: string[]): number | null {
+  const result = spawnSync('git', args);
+
+  if (result.error) {
+    throw result.error;
+  }
+
+  return result.status;
+}
+
+async function createDependencyBumpPR(name: string, currentVersion: string | AlpineLimaISOVersion, latestVersion: string | AlpineLimaISOVersion): Promise<void> {
+  const title = `rddepman: bump ${ name } from ${ currentVersion } to ${ latestVersion }`;
+  const branchName = `rddepman-bump-${ name }-from-${ currentVersion }-to-${ latestVersion }`;
+
+  await getOctokit().rest.pulls.create({
+    owner: GITHUB_OWNER,
+    repo:  GITHUB_REPO,
+    title,
+    base:  MAIN_BRANCH,
+    head:  branchName,
+  });
+}
+
+async function checkDependencies(): Promise<void> {
+  // load current versions of dependencies
+  const depVersionsPath = path.join('src', 'assets', 'dependencies.yaml');
+  const currentVersions = await readDependencyVersions(depVersionsPath);
+
+  // Get a list of dependencies for which:
+  // a) current version !== latest version
+  // b) there is not already a PR (open or closed) to perform this exact bump
+  const versionUpdates: VersionComparison[] = [];
+  const promises = dependencies.map(async(dependency) => {
+    const latestVersion = await dependency.getLatestVersion();
+    const currentVersion = currentVersions[dependency.name as keyof DependencyVersions];
+    const name = dependency.name;
+
+    if (JSON.stringify(currentVersion) === JSON.stringify(latestVersion)) {
+      console.log(`Dependency "${ name }" is at latest version "${ JSON.stringify(currentVersion) }".`);
+
+      return;
+    }
+
+    // try to find PR for this combo of name, current version and latest version
+    const branchName = `rddepman-bump-${ name }-from-${ currentVersion }-to-${ latestVersion }`;
+
+    try {
+      const response = await getOctokit().rest.pulls.list({
+        owner: GITHUB_OWNER, repo: GITHUB_REPO, base: branchName,
+      });
+      const prs = response.data;
+
+      if (prs.length === 0) {
+      } else if (prs.length === 1) {
+        console.log(`Found PR that bumps dependency "${ name }" from "${ currentVersion }" to "${ latestVersion }".`);
+      } else {
+        throw new Error(`Found multiple branches that bump dependency "${ name }" from "${ currentVersion }" to "${ latestVersion }".`);
+      }
+    } catch (error: any) {
+      if (error.status === 404) {
+        console.log(`Could not find PR that bumps dependency "${ name }" from "${ currentVersion }" to "${ latestVersion }". Creating...`);
+        versionUpdates.push({
+          name, currentVersion, latestVersion,
+        });
+      } else {
+        throw error;
```

Comment:
I guess Octokit does retry on throttling, so we don't have to?

Reply:
I don't think it does retry on throttling. Is there a need for doing retries? It seems to me that we can wait 24 hours for the workflow to run again. Or create the PR manually. Or run the workflow manually?

Reply:
It'd be nice to _not_ get mail about a failing action run because we ran out of API access.

I'd be okay with exiting on HTTP 429 (too many requests) gracefully, but that runs into the danger of us not noticing that it never actually passed.  Unfortunately GitHub Actions doesn't have a state for "failed, but do it silently".